lib/manageiq_performance/middlewares/memory.rb

```diff
@@ -0,0 +1,55 @@
+module ManageIQPerformance
+  module Middlewares
+    module Memory
+      private
+
+      def memory_initialize
+      end
+
+      def memory_start(env)
+        Memory.memory_query_data = memory_parse_data
+      end
+
+      def memory_finish(env)
+        Memory.memory_query_data = memory_parse_data(Memory.memory_query_data)
+        save_report env, :memory,
+                    memory_short_form_data,
+                    memory_long_form_data
+      ensure
+        Memory.memory_query_data = nil
+      end
+
+      def memory_long_form_data
+        proc { Memory.memory_query_data }
+      end
+
+      alias_method memory_long_form_data memory_short_form_data
+
+      def self.memory_query_data
+        Thread.current[:miq_perf_request_memory_data]
+      end
+
+      def self.memory_query_data=(value)
+        Thread.current[:miq_perf_request_memory_data] = value
+      end
+
+      def include_memsize?
+        # ManageIQPerformance.config.include_memsize?
+        false
+      end
+
+      def memory_parse_data(old = {})
```

Comment:
I think you will also need to have a:

```ruby
old ||= {}
```

In this method, incase `old` gets passed in as `nil` (have the tests all passing, but one call from `memory_finish` is calling it with `nil`)